core/src/main/java/org/elasticsearch/index/analysis/NamedAnalyzer.java

```diff
@@ -104,4 +106,17 @@ public void setReusableComponents(Analyzer a, String f, TokenStreamComponents c)
             throw new IllegalStateException("NamedAnalyzer cannot be wrapped with a wrapper, only a delegator");
         }
     };
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof NamedAnalyzer)) return false;
+        NamedAnalyzer that = (NamedAnalyzer) o;
+        return Objects.equals(name, that.name);
+    }
```

Comment:
Since an analyzer is essentially code (made up of bundles of other code like tokenizers and token filters), I don't think there is really a way to make them comparable. So I'm not sure there is really anything we can do, the name is the only comparable identifier we have.